main.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"album"
+	"artist"
+	"db"
+	"favourite"
+	"fmt"
+	"net/http"
+	"playlist"
+	"register"
+	"track"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/dgrijalva/jwt-go"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	sqlDb := db.NewSql()
+	defer sqlDb.Close()
+
+	r := setupRouter(sqlDb)
+	_ = r.Run(":8080")
+}
+
+//Middleware
+func AuthorizeJWT() gin.HandlerFunc {
+	return func(c *gin.Context) {
+
+		authHeader := c.GetHeader("Token")
+		tokenString := authHeader
+		token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("Invalid token", token.Header["Token"])
+
+			}
+			return "secretKey", nil
+		})
+
+		if err != nil {
+			c.JSON(403, gin.H{"message": "Your Token has been expired."})
+			return
+		}
+
+		if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
+			if claims["role"] == "Admin" {
+				c.Request.Header.Set("Role", "Admin")
+				return
+
+			} else if claims["role"] == "General" {
+				c.Request.Header.Set("Role", "General")
+				return
+
+			}
+		}
+		fmt.Println(err)
+		c.AbortWithStatus(http.StatusUnauthorized)
+
+	}
+}
+
+//Middleware for user
+func UserAuthorizeJWT() gin.HandlerFunc {
+	return func(c *gin.Context) {
+
+		authHeader := c.GetHeader("Token")
+		tokenString := authHeader
+		token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
```

Comment:
We have to check token is valid or not in one middleware, and in the other, we have to check the role of that token, 
Here work of both middleware is the same, then what is the meaning of taking two same middleware?